Compile error when using Spanned with noms not_line_ending

I have the following two test cases

```rust
use nom::IResult;

use nom::bytes::complete::tag;
use nom::{
    character::complete::not_line_ending,
    combinator::recognize,
};
use nom_span::Spanned;

extern crate nom;
extern crate nom_span;

type Span<'a> = Spanned<&'a str>;

#[test]
fn example() {
    pub fn until_newline(
        input: Span,
    ) -> IResult<Span, &str> {
        let (rest, comment_str) = not_line_ending(input)?;
        Ok((rest, *comment_str))
    }
    let utf8 = Span::new("test\n", true);

    let (utf8_after, _) = until_newline(utf8).unwrap();

    assert_eq!(*utf8_after, "\n");
}

#[test]
fn counter_example() {
    pub fn until_newline(
        input: Span,
    ) -> IResult<Span, &str> {
        let (rest, comment_str) = recognize(tag("test"))(input)?;
        Ok((rest, *comment_str))
    }
    let utf8 = Span::new("test\n", true);

    let (utf8_after, _) = until_newline(utf8).unwrap();

    assert_eq!(*utf8_after, "\n");
}
```

The example test case should read a string until it finds a newline symbol and then return a `&str` for the match.

This fails with the following error

```
   Compiling nom-span v0.1.1 (/usr/home/tant/git/nom-span)
error[E0521]: borrowed data escapes outside of function
  --> tests/example.rs:20:35
   |
18 |         input: Span,
   |         -----
   |         |
   |         `input` is a reference that is only valid in the function body
   |         has type `Spanned<&'1 str>`
19 |     ) -> IResult<Span, &str> {
20 |         let (rest, comment_str) = not_line_ending(input)?;
   |                                   ^^^^^^^^^^^^^^^^^^^^^^
   |                                   |
   |                                   `input` escapes the function body here
   |                                   argument requires that `'1` must outlive `'static`

For more information about this error, try `rustc --explain E0521`.
error: could not compile `nom-span` (test "str") due to previous error
```

What is strange is that this seems to be related to the use of `not_line_ending` since this introduces the `'static' lifetime, by using `Compare<&'static str>`. See https://docs.rs/nom/latest/nom/character/complete/fn.not_line_ending.html

I added the test `counter example` to show that the error only happens with `not_line_ending`.

Can this be caused by this implementation of Compare?

https://github.com/JulesGuesnon/nom-span/blob/b7e70ceb58a7728eee024a1f0eea19da25fa5846/src/lib.rs#L148-L159

I'm sorry I'm a little bit clueless here, since I don't understand how the `'static` lifetime come into the mix.

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -145,15 +145,15 @@ where
     }
 }
 
-impl<T> Compare<T> for Spanned<T>
+impl<T, Comp> Compare<C> for Spanned<T>
 where
-    T: Compare<T>,
+    T: Compare<Comp>,
 {
-    fn compare(&self, t: T) -> nom::CompareResult {
+    fn compare(&self, t: Comp) -> nom::CompareResult {
         self.data.compare(t)
     }
 
-    fn compare_no_case(&self, t: T) -> nom::CompareResult {
+    fn compare_no_case(&self, t: Comp) -> nom::CompareResult {
         self.data.compare_no_case(t)
     }
 }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -145,7 +145,7 @@ where
     }
 }
 
-impl<T, Comp> Compare<C> for Spanned<T>
+impl<T, Comp> Compare<Comp> for Spanned<T>
 where
     T: Compare<Comp>,
 {

